src/app/shared/pipes/order-by.pipe.ts

```diff
@@ -0,0 +1,11 @@
+import { Pipe, PipeTransform } from "@angular/core";
+import { ICourse } from "../../core/index";
+
+@Pipe({
+    name: "orderBy"
+})
+export class OrderByPipe implements PipeTransform {
+    public transform(courses: Array<ICourse>): Array<ICourse> {
+        return courses.sort((courseA: ICourse, courseB: ICourse) => courseB.creationDate.getTime() - courseA.creationDate.getTime());
```

Comment:
it is not a good idea to use sort directly. 
Sort does not return a new array it sorts the source array and mutates it.